src/com/sap/piper/PiperGoUtils.groovy

```diff
@@ -0,0 +1,62 @@
+package com.sap.piper
+
+class PiperGoUtils implements Serializable {
+
+    private static Script steps
+    private static Utils utils
+
+    private static String DELIMITER = '-DeLiMiTeR-'
+
+    PiperGoUtils(Script steps) {
+        this.steps = steps
+        this.utils = new Utils()
+    }
+
+    PiperGoUtils(Script steps, Utils utils) {
+        this.steps = steps
+        this.utils = utils
+    }
+
+    void unstashPiperBin() {
+
+        if (utils.unstash('piper-bin').size() > 0) return
+
+        def libraries = getLibrariesInfo()
+        String version
+        libraries.each {lib ->
+            if (lib.name == 'piper-lib-os') {
+                version = lib.version
+            }
+        }
+
+        def fallbackUrl = 'https://github.com/SAP/jenkins-library/releases/latest/download/piper_master'
+        def piperBinUrl = (version == 'master') ? fallbackUrl : "https://github.com/SAP/jenkins-library/releases/tag/${version}"
+
+        boolean downloaded = downloadGoBinary(piperBinUrl)
+        if (!downloaded) {
+            //Inform that no Piper binary is available for used library branch
+            steps.echo ("Not able to download go binary of Piper for version ${version}")
+            //Fallback to master version & throw error in case this fails
+            if (!downloadGoBinary(fallbackUrl)) steps.error("Download of Piper go binary failed.")
+        }
+        utils.stashWithMessage('piper-bin', 'failed to stash piper binary', 'piper')
+    }
+
+    List getLibrariesInfo() {
+        return new JenkinsUtils().getLibrariesInfo()
+    }
+
+    private boolean downloadGoBinary(url) {
+
+        def response = steps.sh(returnStdout: true, script: "curl --insecure --silent --location --write-out '${DELIMITER}status=%{http_code}' --output ./piper '${url}'")
```

Comment:
What is the story behind the delimiter?

Reply:
... but isn't it sufficient to say `curl --write-out '%{http_code}'`? That returns the `200`, `201`, `403`, `418`, `503` or whatever. With that there is no need to first add a delimeter and remove it afterwards. Also  having the leading `status=` can be omitted. Just the plain status code is sufficient.

That would basically melt down to:

```
def httpStatus = sh "curl ... --write-out '%{http_code}'"
if(httpStatus == '200') {
} else 
}
```
Other option would be to use `curl --fail` and to react on the return code of curl itself. `--fail` results in return codes `!=0` for status codes indicating trouble.

Reply:
There have been longer investigations by @nevskrem which resulted in this solution.
@nevskrem any comment to that?

Reply:
@nevskrem what do you think? What was the reason behind implementing it like that?
If there is a good reason I would suggest to continue as is instead of parking this PR for much longer.

Reply:
@marcusholl @OliverNocon That wasn't actually invented by me but if I remember correctly there is not just the status being written into the output but also additional stuff - at least it was like that in the code were the snippet derives from. Since we over time converted nearly all uses of `curl` to Jenkins http_request it is hard to do forensics on that.
When reading the docs and looking at above scenario I agree that Marcus' code should do the trick already so why not giving it a try?

Reply:
will do so and update the PR then ...

Reply:
has been changed ...